oracle/orchestrator/orchestrator.go

```diff
@@ -1,70 +1,71 @@
 	mmtypes "github.com/skip-mev/slinky/x/marketmap/types"
 )
 
-type (
-	// ProviderOrchestrator is a stateful orchestrator that is responsible for maintaining all of the
-	// providers that the oracle is using. This includes initializing the providers, creating
-	// the provider specific market map, and enabling/disabling the providers based on the
-	// oracle configuration and market map.
-	ProviderOrchestrator struct {
-		mut    sync.Mutex
-		logger *zap.Logger
-
-		// providers is a map of all of the providers that the oracle is using.
-		providers map[string]ProviderState
-
-		// -------------------Oracle Configuration Fields-------------------//
-		//
-		// cfg is the oracle configuration.
-		cfg config.OracleConfig
-		// marketMap is the market map that the oracle is using.
-		marketMap mmtypes.MarketMap
-
-		// -------------------Provider Constructor Fields-------------------//
-		//
-		// apiQueryHandler factory is a factory function that creates API query handlers.
-		apiQueryHandlerFactory types.PriceAPIQueryHandlerFactory
-		// webSocketQueryHandlerFactory is a factory function that creates websocket query
-		// handlers.
-		webSocketQueryHandlerFactory types.PriceWebSocketQueryHandlerFactory
-		// wsMetrics is the web socket metrics.
-		wsMetrics wsmetrics.WebSocketMetrics
-		// apiMetrics is the API metrics.
-		apiMetrics apimetrics.APIMetrics
-		// providerMetrics is the provider metrics.
-		providerMetrics providermetrics.ProviderMetrics
-	}
+// ProviderOrchestrator is a stateful orchestrator that is responsible for maintaining
+// all of the providers that the oracle is using. This includes initializing the providers,
+// creating the provider specific market map, and enabling/disabling the providers based
+// on the oracle configuration and market map.
+type ProviderOrchestrator struct {
+	mut    sync.Mutex
+	logger *zap.Logger
+
+	// -------------------Stateful Fields-------------------//
+	//
+	// providers is a map of all of the providers that the oracle is using.
+	providers map[string]ProviderState
+	// mapper is the market map provider. Specifically this provider is responsible
+	// for making requests for the latest market map data.
+	mapper *mmclienttypes.MarketMapProvider
+
+	// -------------------Oracle Configuration Fields-------------------//
+	//
+	// cfg is the oracle configuration.
+	cfg config.OracleConfig
+	// marketMap is the market map that the oracle is using.
+	marketMap mmtypes.MarketMap
+
+	// -------------------Provider Constructor Fields-------------------//
+	//
+	// priceAPIFactory factory is a factory function that creates price API query handlers.
+	priceAPIFactory types.PriceAPIQueryHandlerFactory
+	// priceWSFactory is a factory function that creates price websocket query handlers.
+	priceWSFactory types.PriceWebSocketQueryHandlerFactory
+	// marketMapperFactory is a factory function that creates market map providers.
+	marketMapperFactory mmclienttypes.MarketMapFactory
```

Comment:
```suggestion
	// marketMapFactory is a factory function that creates market map providers.
	marketMapFactory mmclienttypes.MarketMapFactory
```

Reply:
i prefer `marketMapperFactory` over `marketMapFactory` as market mapper is distinctively different from a market map.